File: app/src/main/java/app/familygem/constant/Format.java
package app.familygem.constant;

/**
 * All of the formats a Date can be displayed as.
 */
public class Format {
    public static final String[] PATTERNS = {"d MMM yyy", "d M yyy", "MMM yyy", "M yyy", "d MMM", "yyy", ""};
    public static final String D_M_Y = PATTERNS[0]; // 31 JAN 2000
    public static final String D_m_Y = PATTERNS[1]; // 31 1 2000
    public static final String M_Y = PATTERNS[2]; // JAN 2000
    public static final String m_Y = PATTERNS[3]; // 1 2000
    public static final String D_M = PATTERNS[4]; // 31 JAN
    public static final String Y = PATTERNS[5]; // 2000
    public static final String OTHER = PATTERNS[6]; // Any empty or invalid date
}

File: app/src/main/java/app/familygem/constant/Format.kt
package app.familygem.constant

/** All the formats a LocalDate can be displayed as. */
enum class Format(val pattern: String) {
    D_M_Y("d MMM yyy"), // 31 JAN 2000
    D_m_Y("d M yyy"), // 31 1 2000
    M_Y("MMM yyy"), // JAN 2000
    m_Y("M yyy"), // 1 2000
    D_M("d MMM"), // 31 JAN
    D("d"), // 31
    M("MMM"), // JAN
    Y("yyy"), // 2000
    OTHER("") // Empty or invalid date
}
